torax/_src/imas_tools/input/core_profiles.py

```
@@ -109,3 +123,33 @@ def profile_conditions_from_IMAS(
       "normalize_n_e_to_nbar": False,
       "v_loop_lcfs": v_loop_lcfs,
   }
+
+def _validate_core_profiles_ids_for_profile_conditions(
+    ids: ids_toplevel.IDSToplevel,
+):
+  """Checks that all expected attributes exist in the IDS."""
+  profiles_1d = ids.profiles_1d
+  global_quantities = ids.global_quantities
+  logged_fields = []
+  for field in _PROFILE_CONDITIONS_REQUIRED_FIELDS["gloabl_quantities"]:
+    if not getattr(global_quantities, field):
+      # Warning or critical ?
+      logging.critical(
```

I believe we'd want to raise an error here with an informative error message that the expected field is missing.

---

Not entirely sure about this. For geometry loader I think we'd want to raise an error but with core_profiles not necessarily. There might be cases where we want to load partially some profiles but not all from an IDS, like only electrons ones or ions ones and get the rest from elsewhere / prescribe it. In this case, it does not matter that the other fields are empty in the input IDS. Do you agree ?